gammapy/makers/map.py

```diff
@@ -75,6 +78,35 @@ def make_counts(geom, observation):
         counts = Map.from_geom(geom)
         counts.fill_events(observation.events)
         return counts
+    
+    @staticmethod
+    def make_counts_rad_max(geom, observation):
+        """Extract the counts using for the ON region size the values in the `RAD_MAX_2D` table"""
+        if observation.rad_max is None:
+            raise TypeError("the IRF for this observation does not include a RAD_MAX_2D table")
```

Comment:
I don't think this is a `TypeError`. `TypeError` is meant for cases like a function expecting an `int` but getting a `str`.

This is either a `ValueError` or maybe, also in regard to https://github.com/gammapy/gammapy/issues/3666, we should create a ` RequiredIRFMissing` or similar exception in gammapy.